### ISSUE TITLE
Login post-registration failing

### ISSUE THREAD

**Issue by mekarpeles** [internetarchive/openlibrary#8863]
#### Evidence / Screenshot

When a patron registers for an account, sometimes they see "no account exists". Sometimes when they re-click login, it them works / created and linked their Archive.org account with Open Library.

### Reproducing the bug

1. Go to ...
2. Do ...

* Expected behavior:
* Actual behavior:

### Context

- Browser (Chrome, Safari, Firefox, etc):
- OS (Windows, Mac, etc):
- Logged in (Y/N): Y
- Environment (prod, dev, local): prod

**Comment by jimchamp** [comment]
@mekarpeles or @seabelis, are there any more details?  Is this happening to patrons who have created an IA account first, or an OL account?  Is this happening for both cases?

Wasn't there another similar issue that was opened recently?

**Comment by seabelis** [comment]
@jimchamp I think the issues may be connected. I think another problem is that they create the account but don't know they have to verify through an email link. They try to sign in with an unverified account and we don't tell them that's the reason they can't sign in.

**Comment by jimchamp** [comment]
I wasn't able to reproduce this issue.  I think that the next steps should be adding some instrumentation to our login and registration paths.  Would be good to capture the exact error that is occurring whenever authentication fails, and maybe the stacktrace.

Some discussion is needed in order to nail down the details.

**Comment by cdrini** [comment]
As a first stab for instrumentation, adding the login errors to sentry would be a good place to give us some observability/transparency. We'll likely need a `logger.exception` where that happens.

**Comment by mekarpeles** [comment]
This may be somewhat tricky on production because we whole-sale sample at something like 2% of web nodes.

We may be able to use `scope` to create a temporary context with 100% sampling for specific errors like login/registration (chatgpt says something like this):
```
import sentry_sdk

# Initialize Sentry with the default sample rate
sentry_sdk.init(
    dsn="your_sentry_dsn",
    traces_sample_rate=0.05  # Default sample rate of 5%
)

def capture_specific_error():
    try:
        # Code that might raise an exception
        raise ValueError("An example error")
    except ValueError as e:
        # Capture this specific error with 100% sampling rate
        with sentry_sdk.configure_scope() as scope:
            scope.sample_rate = 1.0  # Override sample rate to 100%
            sentry_sdk.capture_exception(e)

# Somewhere in your application
capture_specific_error()
```

**Comment by mekarpeles** [comment]
I think this may be closed due to @rebecca-shoptaw's invisible recaptcha fixes #9466
